app/src/models/status.ts

```diff
@@ -138,14 +163,19 @@ export class FileChange {
   /**
    * @param path The relative path to the file in the repository.
    * @param status The status of the change to the file.
-   * @param oldPath The original path in the case of a renamed file.
    */
   public constructor(
     public readonly path: string,
-    public readonly status: AppFileStatus,
-    public readonly oldPath?: string
+    public readonly status: AppFileStatus
   ) {
-    this.id = `${this.status}+${this.path}`
+    if (
+      status.kind === AppFileStatusKind.Renamed ||
+      status.kind === AppFileStatusKind.Copied
+    ) {
+      this.id = `${this.status.kind}+${this.path}+${status.oldPath}`
```

Comment:
what's the significance of this change? is this a bugfix?

Reply:
This was another case that was improved by `oldPath` being moved into `AppFileStatus` - we can generate a better id for these files.